### PR TITLE
Leveraged Strategy Fixtures

### PR DESCRIPTION
**Pull request by 0xrin1**
- Reuse code between test & local deployments by using abstract contract
- Add fixture support to local deployment
  - Execute deposit, harvest & redeem functions

### CODE REVIEW

File: script/DeployLeveragedEthLocal.s.sol [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        deal(address(weth), alice, 100e18);
+        deal(address(weth), bob, 100e18);
+    }
+
+    function deposit(address from, address to, uint256 amount) internal {
+        console2.log("depositing", from);
+
+        vm.startPrank(from);
+        weth.approve(address(to), type(uint256).max);
+        wethContract.deposit(amount, from);
```
Comment (coolhill) [inline script/DeployLeveragedEthLocal.s.sol:61]:
```suggestion
        weth.approve(address(wethContract), type(uint256).max);
        wethContract.deposit(amount, to);
```

Comment (coolhill) [reply script/DeployLeveragedEthLocal.s.sol:61]: otherwise `to` doesn't receive anything

Comment (coolhill) [reply script/DeployLeveragedEthLocal.s.sol:61]: oh nevermind you used weird naming further up

**Comment by 0xrin1** [comment]
i just did for weth for now. can add usdc but maybe as a second PR better so we can start using this without checking out into another branch

**Review by coolhill — APPROVED** [review]

**Pull request merged**